tests/specs/examples/sum-to-n-foundry-spec.k

```diff
@@ -0,0 +1,76 @@
+requires "lemmas/lemmas.k"
+requires "infinite-gas.md"
+
+module VERIFICATION
+    imports LEMMAS
+    imports INFINITE-GAS
+
+    rule N xorInt maxUInt256 => maxUInt256 -Int N 
+    requires #rangeUInt(256, N)
+    [simplification]
+
+    rule A *Int B <=Int C => A <=Int C /Int B
+    requires 0 <Int B
+     andBool 0 <=Int A
+     andBool 0 <=Int C
+    [simplification, concrete(B, C)]
```

Comment:
What about this claim breaks without this lemma? I'm curious why we'd want to switch from a multiplication to a division as well. I'm also still having a hard time proving this to be true. It's obvious in the `Real` domain, but in `Int` is it true?

Reply:
I'm rerunning it to get a more detailed answer, but it was something to do with the gas calculation, where it was unable to show `N *Int 178` was less than the current amount of gas, and this was happening repeatedly when gas was deducted so I needed a more general lemma.